app/src/main/java/com/example/moodtrackerproject/ui/health/EditHealthViewModel.kt

```diff
@@ -16,14 +18,32 @@ class EditHealthViewModel : BaseViewModel<EditHealthProps>() {
     override fun map(appState: AppState, action: MviAction?): EditHealthProps {
         val state = appState.editHealthState
         return EditHealthProps(
-            waterNum = state.waterNum,
-            stepsNum = state.stepsNum,
-            sleepNum = state.sleepNum,
-            kcalNum = state.kcalNum
+            action = action as? EditHealthScreenActions,
+            listHealth =
+            EditHealthItemProps(
+                water = DataBaseRepository.getHealth()[0].water,
```

Comment:
When you call `DataBaseRepository.getHealth()` in every row it makes reading from prefs and parsing every single time, do you understand that it is resources-consuming operation? Please, extract it to local variable